.github/workflows/main.yaml

```diff
@@ -4,6 +4,10 @@ on:
     branches:
       - "main"
 
+permissions:
+  contents: read
+  packages: write
```

Comment:
This + using `${{ secrets.GITHUB_TOKEN }}` below avoids that people have to create a custom token and is generally more secure as those credentials are only short-living.